modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java

```diff
@@ -369,7 +361,6 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
                 DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(syncmlDocument.
                         getHeader().getSource().getLocURI());
                 Device existingDevice = WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier);
```

Comment:
possible NPE scenario if we do not get the service here. better checking it first and invoke .getDevice